homework_4/betsapi/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,155 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	storagemodels "github.com/superbet-group/code-cadets-2021/homework_4/betsapi/internal/infrastructure/sqlite/models"
+)
+
+// BetRepository provides methods that operate on bets SQLite database.
+type BetRepository struct {
+	dbExecutor DatabaseExecutor
+}
+
+// NewBetRepository creates and returns a new BetRepository.
+func NewBetRepository(dbExecutor DatabaseExecutor) *BetRepository {
+	return &BetRepository{
+		dbExecutor: dbExecutor,
+	}
+}
+
+// readBet tries to read a single bet from given sql rows.
+// If sql row has no more rows to scan, an error is returned.
+func (b *BetRepository) readBet(row *sql.Rows) (storagemodels.Bet, error) {
+	var id string
+	var customerId string
+	var status string
+	var selectionId string
+	var selectionCoefficient int
+	var payment int
+	var payoutSql sql.NullInt64
+
+	err := row.Scan(&id, &customerId, &status, &selectionId, &selectionCoefficient, &payment, &payoutSql)
+	if err != nil {
+		return storagemodels.Bet{}, err
+	}
+
+	var payout int
+	if payoutSql.Valid {
+		payout = int(payoutSql.Int64)
+	}
+
+	return storagemodels.Bet{
+		Id:                   id,
+		CustomerId:           customerId,
+		Status:               status,
+		SelectionId:          selectionId,
+		SelectionCoefficient: selectionCoefficient,
+		Payment:              payment,
+		Payout:               payout,
+	}, nil
+}
+
+// GetBetById fetches a bet from the database and returns it. The second returned value indicates
+// whether the bet exists in DB. If the bet does not exist, an error will not be returned.
+func (b *BetRepository) GetBetById(ctx context.Context, id string) (storagemodels.Bet, bool, error) {
+	storageBet, err := b.queryGetBetById(ctx, id)
+	if err == sql.ErrNoRows {
+		return storagemodels.Bet{}, false, nil
+	}
+	if err != nil {
+		return storagemodels.Bet{}, false, errors.Wrap(err, "bet repository failed to get a bet with id "+id)
+	}
+
+	exists := storageBet != storagemodels.Bet{}
+	return storageBet, exists, nil
+}
+
+func (b *BetRepository) queryGetBetById(ctx context.Context, id string) (storagemodels.Bet, error) {
+	row, err := b.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id='"+id+"';")
```

Comment:
You're setting yourself up for a possible SQL injection attack here. `QueryContext` and `Query` methods can sanitise the arguments for you. I believe the placeholder syntax depends on the underlying DB driver but it's most probably a `?`. Like this:
```suggestion
	row, err := b.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id=?;", id)
```

Same comment for other queries.